WikiSentiment/WikiData/DailyCollection.cs

```diff
@@ -0,0 +1,135 @@
+            {
+                countrydailydict = regionalCollections,
+                featuredlist = featuredCountries
+            }; ;
+        }
+
+        /// <summary>
+        /// Updates old colection with new content, compiles new featured list
+        /// </summary>
+        /// <param name="_base"></param>
+        /// <param name="_newAdditions"></param>
+        /// <returns></returns>
+        public static DailyCollection UpdateGiven(DailyCollection _base, DailyCollection _newAdditions)
+        {
+            var newCollection = new Dictionary<string, LanguageCollection>(_base.countrydailydict);
+
+            //override old data with new ones
+            foreach(string iCountry in _newAdditions.countrydailydict.Keys)
+            {
+                newCollection[iCountry] = _newAdditions.countrydailydict[iCountry];
+            }
+            var featured = getFeaturedCountries(newCollection, featureArticles);
+
+            return new DailyCollection()
+            {
+                countrydailydict = newCollection,
+                featuredlist = featured
+            };
+        }
+
+        /// <summary>
+        /// Get ordered list of languages with the top viewed (proportionally) articles
+        /// </summary>
+        /// <param name="_collection"></param>
+        /// <param name="_featuredAmount">amount of languages to keep</param>
+        /// <returns>list of two letter language codes</returns>
+        static List<string> getFeaturedCountries(
+            Dictionary<string, LanguageCollection> _collection, int _featuredAmount)
+        {
+            //compile the list of country codes and viewership scores for top articles
+            var languageTopArticlesData = new List<(string countrycode, float percentage)>();
+            foreach (string iLanguage in _collection.Keys)
+            {
+                languageTopArticlesData.Add((iLanguage,
+                    (100f * _collection[iLanguage].articles[0].vws) / _collection[iLanguage].totalviews));
+            }
+
+            //order data in the descending order
+            languageTopArticlesData = (languageTopArticlesData.OrderBy(data => data.percentage)).Reverse().ToList();
+
+            var amountToFeature = Math.Min(_featuredAmount, _collection.Count);
+            List<string> result = new List<string>();
+
+            //return countrycodes
+            for (int i = 0; i < amountToFeature; i++)
+                result.Add(languageTopArticlesData[i].countrycode);
+            
+            return result;
+        }
+
+        public string ToJSON()
+        {
+            return JsonSerializer.Serialize(this, new JsonSerializerOptions()
+            {
+                Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+                WriteIndented = false
+            });
+        }
+
+        public bool IsValid()
+        {
+            return countrydailydict != null && countrydailydict.Count > 0;
+        }
```

Comment:
Not always avoidable, but its better if you can simply avoid creating invalid objects.

Reply:
This is part of a greater theme of "Instead of having to check if something is in a valid state, make the invalid state impossible". 

In practice this can be tricky.